.github/workflows/functional-tests.yml

```
@@ -60,58 +173,97 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
+
+      - name: Download Community Frontend Build Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: community-frontend-build
+          path: ${{ env.frontend-directory }}/
+
       - name: Set up python ${{ matrix.python-version }}
         uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)